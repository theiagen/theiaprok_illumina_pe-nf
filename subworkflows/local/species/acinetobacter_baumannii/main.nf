include { KAPTIVE  } from '../../../../modules/local/species_typing/acinobacter/kaptive/main'
include { ABRICATE } from '../../../../modules/local/gene_typing/drug_resistance/abricate/main'
workflow ACINETOBACTER_SPECIES {

    take:
    ch_acinetobacter // Channel of tuples [meta, assembly, reads, species]

    main:

    ch_samples = ch_acinetobacter.map { meta, assembly, reads, species -> [meta, assembly] }

    ch_versions = Channel.empty()
    ch_kaptive_results = Channel.empty()
    ch_abricate_results = Channel.empty()

    KAPTIVE (
        ch_samples,
        params.kaptive_start_end_margin ?: 10,
        params.kaptive_min_percent_coverage ?: 90.0,
        params.kaptive_min_percent_identity ?: 80.0,
        params.kaptive_low_gene_percent_identity ?: 95.0
    )
    ch_kaptive_results = KAPTIVE.out.tables
    ch_versions = ch_versions.mix(KAPTIVE.out.versions)

    ABRICATE (
        ch_samples,
        "AcinetobacterPlasmidTyping",
        params.abricate_abaum_min_percent_identity ?: 95,
        params.abricate_abaum_min_percent_coverage ?: 90
    )
    ch_abricate_results = ABRICATE.out.results
    ch_versions = ch_versions.mix(ABRICATE.out.versions)

    emit:
    kaptive_results = ch_kaptive_results
    abricate_results = ch_abricate_results
    
    versions = ch_versions
}