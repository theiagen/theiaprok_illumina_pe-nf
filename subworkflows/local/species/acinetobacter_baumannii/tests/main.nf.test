nextflow_workflow {

    name "Test Subworkflow ACINETOBACTER_SPECIES"
    script "../main.nf"
    workflow "ACINETOBACTER_SPECIES"

    // Currently utilizes local test data, needs to be replaced with remote data
    test("acinetobacter_species - abaumannii characterization") {
        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test_abaum', single_end:false ],
                    file("gs://fc-c37b6bc6-b033-4ef1-b8aa-b8c1d25524d4/submissions/9b83240f-3549-42f5-8960-d6d04ed6555b/theiaprok_illumina_pe/e21df9da-e052-4eee-b8c5-ddcb57390962/call-digger_denovo/digger_denovo/bdeac325-74cf-4d17-b5a6-dadc2910b58d/call-filter_contigs/cacheCopy/20012105104_filtered_contigs.fasta"),
                    [
                        file("gs://theiagen-large-public-files-rp/terra/theiaprok-files/theiaprok_validation/20012105104_R1.fastq.gz"),
                        file("gs://theiagen-large-public-files-rp/terra/theiaprok-files/theiaprok_validation/20012105104_R2.fastq.gz")
                    ],
                    "Acinetobacter baumannii"
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.kaptive_results,
                    workflow.out.abricate_results,
                    workflow.out.versions).match() }
            )
        }
    }
}