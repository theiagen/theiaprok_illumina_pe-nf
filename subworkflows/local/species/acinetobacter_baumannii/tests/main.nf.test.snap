{
    "acinetobacter_species - abaumannii characterization": {
        "content": [
            [
                
            ],
            [
                [
                    {
                        "id": "test_abaum",
                        "single_end": false
                    },
                    "test_abaum_abricate_hits.tsv:md5,997119371d441dd564173ebf40abc22b"
                ]
            ],
            [
                "versions.yml:md5,48d9c953e7b490c7d631409b0d4d1117"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.8"
        },
        "timestamp": "2025-07-02T18:35:41.378060555"
    }
}